db.js

```diff
@@ -232,9 +232,10 @@ export class ActivityPubDB extends EventTarget {
     }
 
     // Implement additional filtering logic if needed based on other criteria (like time ranges or tags)
-    // For example:
-    // notes.filter(note => note.published >= criteria.startDate && note.published <= criteria.endDate);
-    return notes.sort((a, b) => b.published - a.published) // Sort by published date in descending order
+    const sortedNotes = notes.sort((a, b) => b.published - a.published)
```

Comment:
Instead of getting the full list and slicing we should have the transaction use an index that's sorted by published time. From there we should do a for await loop over the index and keep track of the number of items to count. If you need help figuring out how to do that please book a slot on my cal.

Reply:
https://github.com/hyphacoop/reader.distributed.press/pull/8